src/typeahead/getMapSuggestions.ts

```diff
@@ -0,0 +1,229 @@
+  // If there is a parent layer only the child has to be enabled, otherwise all layers.
+  const enabledSelection = parentLayer ? buildSelectionIds(layer, collection) : selectionIds
+
+  return buildSuggestion('map-layer', layer.title, selectionIds, enabledSelection)
+}
+
+function mapCollectionToContent(collection: EnrichedMapCollection): TypeAheadSuggestionContent {
+  const selectionIds = collection.mapLayers
+    .map(layer => buildSelectionIds(layer, collection))
+    .flat()
+
+  return buildSuggestion('map-collection', collection.title, selectionIds)
+}
+
+interface EnrichedMapCollection extends MapCollection {
+  mapLayers: EnrichedMapLayer[]
+}
+
+/**
+ * Enriches map collections with additional data for layers.
+ *
+ * @param collections The collections to enrich with additional data.
+ */
+function enrichMapCollections(collections: MapCollection[]): EnrichedMapCollection[] {
+  return collections.map(collection => {
+    const layerIds = collection.mapLayers.map(layer => layer.id)
+    const matchedLayers: MapLayer[] = []
+
+    for (let layerId of layerIds) {
+      const matchingLayer = layersById.get(layerId)
+
+      if (matchingLayer) {
+        matchedLayers.push(matchingLayer)
+      } else {
+        console.error(
+          `Unable to find matching layer with id '${layerId}' for map collection with id ${collection.id}.`,
+        )
+      }
+    }
+
+    return {
+      ...collection,
+      mapLayers: enrichMapLayers(matchedLayers),
+    }
+  })
+}
+
+interface EnrichedMapLayer extends MapLayer {
+  themes: Theme[]
+}
+
+/**
+ * Enriches map layers with additional data for themes.
+ *
+ * @param layers The layers to enrich with additional data.
+ */
+function enrichMapLayers(layers: MapLayer[]): EnrichedMapLayer[] {
+  return layers.map(layer => {
+    const themeIds = layer.meta?.themes || []
+    const matchedThemes: Theme[] = []
+
+    for (let themeId of themeIds) {
+      const matchingTheme = themesById.get(themeId)
+
+      if (matchingTheme) {
+        matchedThemes.push(matchingTheme)
+      } else {
+        console.error(
+          `Unable to find matching theme with id '${themeId}' for map layer with id ${layer.id}.`,
+        )
+      }
+    }
+
+    return {
+      ...layer,
+      themes: matchedThemes,
+    }
+  })
+}
+
+/**
+ * Builds a suggestion with all corresponding fields.
+ *
+ * @param type The type of suggestion (e.g `'map-collection' or `'map-layer'`)
+ * @param title The title to display for this suggestion.
+ * @param layerIds The map layers which belong to this suggestion, used to build a deep link.
+ * @param enabledLayers The map layers that should be enabled, if not specified all map layers will be enabled.
+ */
+function buildSuggestion(
+  type: string,
+  title: string,
+  layerIds: string[],
+  enabledLayers = layerIds,
+): TypeAheadSuggestionContent {
+  const layers = layerIds.map(id => `${id}:${enabledLayers.includes(id) ? '1' : '0'}`).join('|')
```

Comment:
are we sure we want to put this logic here? although we have more dataportaal specific code in this repo, it would seem a better idea to move this to atlas